src/ui/internal/hooks/useCountdownChallengePeriod.tsx

```diff
@@ -1,152 +1,127 @@
-    React.useState<string>("Confirm Challenge");
-  const [disableButton, setDisableButton] = React.useState(false);
-  const [labelChallengePeriod, setLabelChallengePeriod] =
-    React.useState("Challenge Period");
-  const [countdown, setCountdown] = React.useState("...");
-
-  const [shouldWaitOtherParty, setShouldWaitOtherParty] = React.useState(false);
-
-  const setupCountdown = (date: Date, periodStarted = true) => {
-    const _countdown = countdownChallengePeriod(date);
-
-    if (_countdown === "-") {
-      setCountdown("expired");
-    } else {
-      if (periodStarted) {
-        setCountdown(_countdown);
-      } else {
-        setCountdown(`Starts in ${_countdown.replace(" remaining", "")}`);
-      }
-    }
+import { countdownChallengePeriod } from "helpers/countdownChallengePeriod";
+import React from "react";
+
+const hasExpired = (period: Date) => period && Date.now() > period.getTime();
+const getCountdown = (date, prefix) =>
+  prefix + " " + countdownChallengePeriod(date).replace(" remaining", "");
+
+const getChallengedBy = (data) => {
+  return data.status.latestChallengeBy === null
+    ? "buyer"
+    : data.status.latestChallengeBy;
+};
+
+const extractChallengeData = (data) => ({
+  start: data.challengePeriodStart,
+  end: data.challengePeriodEnd,
+  connectedUser: data.connectedUser,
+  neverChallenged: data.status.latestChallengeBy === null,
+  challengedBy: getChallengedBy(data),
+});
+
+export const useCountdownChallengePeriod = (escrowData) => {
+  const data = React.useRef<any>(escrowData);
+  const challenge = React.useRef<any>(null);
+  const timer = React.useRef<NodeJS.Timer>();
+  const countdown = React.useRef<any>("...");
+  const [canChallenge, setCanChallenge] = React.useState<boolean>(false);
+
+  const startExpired = hasExpired(challenge.current?.start);
+  const endExpired = hasExpired(challenge.current?.end);
+
+  const updateChallenge = (newData) => {
+    data.current = newData;
+    challenge.current = {
+      ...challenge.current,
+      ...extractChallengeData(newData),
+    };
+    resetCountdown();
   };
 
-  const isExpired = (period: Date) => {
-    return countdownChallengePeriod(period) === "-";
+  const startChallenge = () => {
+    setCanChallenge(false);
+    startCountdown();
   };
 
-  const count = (period: Date, periodStarted: boolean) => {
-    const timer = setInterval(() => {
-      setupCountdown(period, periodStarted);
-    }, 1000);
+  const startCountdown = () => {
+    timer.current && resetCountdown();
+    timer.current = setInterval(() => {
+      const { end, start, challengedBy, neverChallenged, connectedUser } =
+        challenge.current;
+      const challengedByYou = challengedBy === connectedUser;
+      let prefix;
+      let date;
+
+      if (startExpired || neverChallenged) {
+        date = end;
+        prefix = "Expires in";
+      } else {
+        date = challengedByYou ? end : start;
+        prefix = challengedByYou ? "Expires in" : "Begins in";
```

Comment:
Starts in 
If Tomas is ok with Begins, that's ok.
Just saying this because of the figma modals ;)